src/pal/src/file/file.cpp

```diff
@@ -4971,27 +4885,31 @@ Input parameters:
 source  = path to the file on input, path to the file with all 
           symbolic links traversed on return
 
-Note: Assumes the maximum size of the source is MAX_LONGPATH
-
 Return value:
     TRUE on success, FALSE on failure
 --*/
-BOOL FILEGetFileNameFromSymLink(char *source)
+BOOL FILEGetFileNameFromSymLink(PathCharString& source)
```

Comment:
This function doesn't really work correctly, but that's independent of this feature. I've created issue #3450 for it.